voyager.go

```diff
@@ -0,0 +1,327 @@
+package voyager
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/concourse/voyager/runner"
+	multierror "github.com/hashicorp/go-multierror"
+	_ "github.com/lib/pq"
+)
+
+type Migrator interface {
+	CurrentVersion() (int, error)
+	SupportedVersion() (int, error)
+	Migrate(version int) error
+	Up() error
+	Migrations() ([]migration, []int, error)
+}
+
+func NewMigrator(db *sql.DB, lockID int, source Source, migrationsRunner runner.MigrationsRunner, adapter SchemaAdapter) Migrator {
+	return &migrator{
+		db,
+		lockID,
+		lager.NewLogger("migrations"),
+		source,
+		migrationsRunner,
+		adapter,
+		&sync.Mutex{},
+	}
+}
+
+type migrator struct {
+	db                 *sql.DB
+	lockID             int
+	logger             lager.Logger
+	source             Source
+	goMigrationsRunner runner.MigrationsRunner
+	adapter            SchemaAdapter
+	*sync.Mutex
+}
+
+//go:generate counterfeiter . SchemaAdapter
+type SchemaAdapter interface {
+	MigrateFromOldSchema(*sql.DB) (int, error)
+	MigrateToOldSchema(*sql.DB, int) error
+	OldSchemaLastVersion() int
+}
+
+func (m *migrator) SupportedVersion() (int, error) {
+	matches := []migration{}
+
+	assets := m.source.AssetNames()
+
+	var parser = NewParser(m.source)
+	for _, match := range assets {
+		if migration, err := parser.ParseMigrationFilename(match); err == nil {
+			matches = append(matches, migration)
+		}
+	}
+	sortMigrations(matches)
+	return matches[len(matches)-1].Version, nil
+}
+
+func (m *migrator) CurrentVersion() (int, error) {
+
+	var migrationHistoryExists bool
+	err := m.db.QueryRow("SELECT EXISTS (SELECT 1 FROM information_schema.tables where table_name='migrations_history')").Scan(&migrationHistoryExists)
```

Comment:
This value for `err` is never checked. 

If `m.db.QueryRow(...).Scan()` will only ever error with unhelpful errors, I'm 100% a fan of assigning to the blank identifier (`_ = ...`) to call out that the value is ignored with good reason.  

Otherwise there may be a need to check the error, maybe comparing against the `sql` package errors (https://golang.org/pkg/database/sql/#pkg-variables) to - for example - see if the error is `sql. ErrNoRows` and continuing on.